src/components/output/custom-els/MultiPanel/index.ts

```diff
@@ -0,0 +1,220 @@
+    const selectedEl = document.activeElement;
+    const heading = getClosestHeading(selectedEl);
+
+    // if keydown event is not on heading element, ignore
+    if (!heading) return;
+
+    // don’t handle modifier shortcuts used by assistive technology.
+    if (event.altKey) return;
+
+    let newHeading;
+    switch (event.key) {
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        newHeading = this._prevHeading();
+        break;
+
+      case 'ArrowRight':
+      case 'ArrowDown':
+        newHeading = this._nextHeading();
+        break;
+
+      case 'Home':
+        newHeading = this._firstHeading();
+        break;
+
+      case 'End':
+        newHeading = this._lastHeading();
+        break;
+
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        this._expand(heading);
+        break;
+
+      // Any other key press is ignored and passed back to the browser.
+      default:
+        return;
+    }
+
+    event.preventDefault();
+    selectedEl.setAttribute('tabindex', '-1');
+    if (newHeading) {
+      newHeading.setAttribute('tabindex', '0');
+      newHeading.focus();
+    }
+  }
+
+  private _expand(heading: Element) {
+    if (!heading) return;
+    const content = heading.nextElementSibling;
+
+    // if there is no content, nothing to expand
+    if (!content) return;
+
+    // toggle expanded and aria-expanded attributes
+    if (content.hasAttribute('expanded')) {
+      content.removeAttribute('expanded');
+      content.setAttribute('aria-expanded', 'false');
+    } else {
+      content.setAttribute('expanded', '');
+      content.setAttribute('aria-expanded', 'true');
+    }
+  }
+
+  // children of multi-panel should always be even number (heading/content pair)
+  private _childrenChange() {
+    let preserveTabIndex : boolean = false;
+    let heading = this.firstElementChild;
+
+    while (heading) {
+      const content = heading.nextElementSibling;
+      const randomId = Math.random().toString(36).substr(2, 9);
+
+      // if at the end of this loop, runout of element for content,
+      // it means it has odd number of elements. log error and set heading to end the loop.
+      if (!content) {
+        console.error('<multi-panel> requires an even number of element children.');
+        heading = null;
+        continue;
+      }
+
+      // When odd number of elements were inserted in the middle,
+      // what was heading before may become content after the insertion.
+      // Remove classes and attributes to prepare for this change.
+      if (heading.classList.contains('panel-content')) {
+        heading.classList.remove('panel-content');
+      }
+      if (content.classList.contains('panel-heading')) {
+        content.classList.remove('panel-heading');
+      }
+      if (heading.hasAttribute('expanded') && heading.hasAttribute('aria-expanded')) {
+        heading.removeAttribute('expanded');
+        heading.removeAttribute('aria-expanded');
+      }
```

Comment:
If I have an expanded section open, but add another section, it'll close the one I had open. That doesn't seem ideal, but I guess it could be filed as a separate issue.

Reply:
I'd like to take this to separate issue. Put some real UI elements inside the panel and see how it should behave along side testing how we should move focus.